packages/core/admin/admin/src/core/apis/Widgets.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable check-file/filename-naming-convention */
+
+import invariant from 'invariant';
+import { To } from 'react-router-dom';
+
+import { Permission } from '../../../../shared/contracts/shared';
+
+import type { MessageDescriptor } from 'react-intl';
+
+type WidgetUID = `plugin::${string}.${string}` | `global::${string}`;
```

Comment:
Might be a bit more verbose, but if you want to be future proof & rely on existing utils, you could use the following
```suggestion
import type { Internal, Utils } from '@strapi/types';

type WidgetUID = Utils.String.Suffix<
  | Internal.Namespace.WithSeparator<Internal.Namespace.Plugin>
  | Internal.Namespace.WithSeparator<Internal.Namespace.Global>,
  string
>;
```

Or even add it to the `Internal.uid` namespace in https://github.com/strapi/strapi/blob/develop/packages/core/types/src/internal/uid.ts (if that make sense to have it defined for broader usage)

Reply:
Nice, we could add it to that file. Custom fields uses the same pattern. Just not sure what the name could be so it could work for both or anything else in the future. wdyt @remidej ?